Infrastructure/SolutionRepository.cs

```diff
@@ -79,9 +81,122 @@ SELECT Count(*) FROM anon_users WHERE user_id = @UserID AND session_id = @Sessio
 		return result > 0;
 	}
 
-	public async Task<bool> InsertSolvedRelation(int userId, int exerciseId)
+	public async Task<LanguageSupport?> GetSolutionLanguageBySession(int languageId, int sessionId)
```

Comment:
Check use - potentially rename from Get to Verify/Check or something similar, as this is intended for internal validation, rather than returning to frontend.